pandas/core/series.py

```diff
@@ -1195,12 +1195,14 @@ def reset_index(self, level=None, drop=False, name=None, inplace=False):
         inplace = validate_bool_kwarg(inplace, 'inplace')
         if drop:
             new_index = com._default_index(len(self))
-            if level is not None and isinstance(self.index, MultiIndex):
+            if level is not None:
                 if not isinstance(level, (tuple, list)):
                     level = [level]
+                level = set(level)
```

Comment:
Why?!

Reply:
`level = set(level)` is not required now. Did not remove it after combining two separate blocks post-testing. Will delete it.